[docs] Add page for `experimentalStyled()` (#26361)
mui-org/material-ui: docs/src/pages/components/box/box.md

<p class="description">The Box component serves as a wrapper component for most of the CSS utility needs.</p>

> The Box component packages [all the style functions](/system/basics/#all-inclusive) that are exposed in `@material-ui/system`.

Reviewer: ```suggestion
The Box component packages [all the style functions](/system/basics/#all-inclusive) that are provided by `@material-ui/system`.
```

> It's created using the [`experimentalStyled()`](/customization/experimental-styled) function of `@material-ui/core/styles`.

Reviewer 2: ```suggestion
It's created using the [`experimentalStyled()`](/customization/experimental-styled/) function of `@material-ui/core/styles`.
```
